test/functional/normal/tab_spec.lua

```diff
@@ -0,0 +1,230 @@
+       .. 'Tab page 2\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 3\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 4\n'
+       .. '    [No Name]',
+       redir_exec('tabs')
+    )
+  end)
+
+  it('switches to earlier after switching to previous tab', function()
+    -- Add three tabs for a total of four
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    -- Switch to the previous (third) tab
+    feed('gT')
+    -- Switch to the earlier (fourth) tab
+    feed(':tabantecedent<cr>')
+
+    eq(   '\n'
+       .. '\n'
+       .. 'Tab page 1\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 2\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 3\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 4\n'
+       .. '>   [No Name]',
+       redir_exec('tabs')
+    )
+  end)
+
+  it('switches to earlier after switching to first tab', function()
+    -- Add three tabs for a total of four
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    -- Switch to the previous (third) tab
+    feed('gT')
+    -- Switch to the first tab
+    feed(':tabfirst<cr>')
+    -- Switch to the earlier (third) tab
+    feed(':tabantecedent<cr>')
+
+    eq(   '\n'
+       .. '\n'
+       .. 'Tab page 1\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 2\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 3\n'
+       .. '>   [No Name]\n'
+       .. 'Tab page 4\n'
+       .. '    [No Name]',
+       redir_exec('tabs')
+    )
+  end)
+
+  it('switches to earlier after numbered tab switch', function()
+    -- Add three tabs for a total of four
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    -- Switch to the second tab
+    feed('2gt')
+    -- Switch to the earlier (fourth) tab
+    feed(':tabantecedent<cr>')
+
+    eq(   '\n'
+       .. '\n'
+       .. 'Tab page 1\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 2\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 3\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 4\n'
+       .. '>   [No Name]',
+       redir_exec('tabs')
+    )
+  end)
+
+  it('switches to earlier after switching to earlier', function()
+    -- Add three tabs for a total of four
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    -- Switch to the second tab
+    feed('2gt')
+    -- Switch to the earlier (fourth) tab
+    feed(':tabantecedent<cr>')
+    -- Switch to the earlier (second) tab
+    feed(':tabantecedent<cr>')
+
+    eq(   '\n'
+       .. '\n'
+       .. 'Tab page 1\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 2\n'
+       .. '>   [No Name]\n'
+       .. 'Tab page 3\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 4\n'
+       .. '    [No Name]',
+       redir_exec('tabs')
+    )
+  end)
+
+  it('does not switch to earlier after closing current tab', function()
+    -- Add three tabs for a total of four
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
```

Comment:
`helpers.command` is better than `feed()` unless you are intentionally testing user input (`feed()`).

Reply:
Fixed to use `command` rather than `feed` except when feeding `g<tab>`.